src/main/java/dev/blaauwendraad/masker/json/MaskingState.java

```diff
@@ -104,23 +104,31 @@ public void backtrackCurrentJsonPath() {
         currentJsonPath.pop();
     }
 
-    /**
-     * Checks if the last segment of the current jsonpath is an array index.
-     */
-    public boolean isInArray() {
-        return !currentJsonPath.isEmpty() && currentJsonPath.peek().offset == -1;
-    }
-
     /**
      * Increments an array index in the last segment of the current jsonpath.
      * Throws {@link java.lang.IllegalStateException} if the last segment is not an array index.
      */
     public void incrementCurrentJsonPathArrayIndex() {
         SegmentReference lastSegment = currentJsonPath.peek();
-        assert isInArray() && lastSegment != null;
+        assert isInArraySegment() && lastSegment != null;
         lastSegment.start++;
     }
 
+    /**
+     * Checks if the current segment of the current jsonpath is an array index.
+     */
+    public boolean isInArraySegment() {
```

Comment:
JavaDoc says "an array index", method says "in array segment"

Reply:
I will fix it.